pkg/operator/controller/gatewayapi/controller_test.go

```diff
@@ -88,6 +122,85 @@ func Test_Reconcile(t *testing.T) {
 			expectDelete:    []client.Object{},
 			expectStartCtrl: false,
 		},
+		{
+			name:                        "unmanaged gateway API CRDs created",
+			gatewayAPIEnabled:           true,
+			gatewayAPIControllerEnabled: true,
+			existingObjects: []runtime.Object{
+				co("ingress"),
+				crd("listenersets.gateway.networking.x-k8s.io"),
+			},
+			existingStatusSubresource: []client.Object{
+				co("ingress"),
```

Comment:
Can you add a comment about why we need an ingress co in both existingObjects and existingStatusSubresource?  Do they represent the same ingress co?

Reply:
> Do they represent the same ingress co?

Yes, they do. I have a comment for `existingStatusSubresource` above which explains that this field is needed if `Reconcile` uses `Status()` calls, which `Reconcile` is doing now.  The reason why it's a dedicated field is because it mimics the fake controller-runtime's mechanics which have a dedicated method for the status subresource: `WithStatusSubresource` (see below).